store/src/directory.rs

```diff
@@ -138,41 +281,126 @@ where
                 Ok(v) if v.is_file() => {}
                 Ok(_) => continue,
             }
+            // skip metadata files if xattr is not supported, we are iterating
+            // the files that have content but filtering based on metadata
+            if !self.xattr_enabled {
+                match path.as_path().extension() {
+                    None => (),
+                    Some(ex) => {
+                        if ex == FDO_METADATA_EX {
+                            continue;
+                        }
+                    }
+                }
+            }
             let mut neqs: HashSet<PathBuf> = HashSet::new();
             for neq in &self.neqs {
                 let (key, expected) = neq;
-                match xattr::get(path.clone(), format_xattr(key)) {
-                    Ok(Some(v)) => {
-                        let matching = expected.iter().zip(&v).filter(|&(a, b)| a == b).count();
-                        if expected.len() != matching {
+
+                if self.xattr_enabled {
+                    match xattr::get(path.clone(), format_xattr(key)) {
```

Comment:
what if we further abstract xattrs enabled or not behind a common trait? we can then have a simple interface for querying them and the underlying implementation will make decisions based on whether xattrs are supported or not

Reply:
I wanted to avoid that in order to touch the minimum number of files, but I guess that switching on a flag is not that clean. Will do that.